lib/quickbooks/model/active_record_scaffold.rb

```diff
@@ -0,0 +1,113 @@
+module Quickbooks
+  module Model
+    module ActiveRecordScaffold
+
+      def self.included(base)
+        base.extend(ClassMethods)
+      end
+
+      def save(options = {})
+        if id.blank?
+          self.class.send(:initialize_service).create(self, options)
+        else
+          self.class.send(:initialize_service).update(self, options)
+        end
+      end
+
+      module ClassMethods
+        def all(query = nil, options = {})
```

Comment:
i'm not sure this quite matches up with active record: http://api.rubyonrails.org/classes/ActiveRecord/Scoping/Named/ClassMethods.html#method-i-all